### DISCUSSION TITLE
Converting heif to Pdf

### DISCUSSION THREAD

**Discussion by abbottmw** [dlemstra/Magick.NET#626] [General] [answered]
Hey all,

Im trying to figure out how to convert an heif file to a pdf.      I have been able to have it create a pdf, and also put each image inside the heif file into a new page of the pdf.   The thing I havent been able to figure out now is the PDF is 16MB in size and also the images in the PDF look like they are missing some color layers, so I may have an RGB/CMYK etc not working correctly.

Is there a way to make the PDF smaller too?

I was looping the inner images to see if i could set quality or maybe other settings to help with the color issue.

Here is my watered down code.

You can dowload the image im using  [here](https://nokiatech.github.io/heif/content/images/season_collection_1440x960.heic)

```
 byte[] result;
            using (var memStream = new MemoryStream())
            {
                using (MagickImageCollection images = new MagickImageCollection("season_collection_1440x960.heic", new MagickReadSettings { Format = MagickFormat.Heic }))
                {

                    foreach (var image in images)
                    {
                        image.Format = MagickFormat.Jpg;
                        image.Quality = 100;
                    }


                    images.Write(memStream, MagickFormat.Pdf);
                }

                result = memStream.ToArray();
            }

            File.WriteAllBytes(@"pdftest.pdf", result);
        }
```

**Comment by dlemstra** [comment, marked as answer]
You can make the PDF smaller by changing the compression method. You will need to decide yourself which one works the best for you:

```C#
foreach (var image in images)
{
    image.Settings.Compression = CompressionMethod.Zip;

    image.Settings.Compression = CompressionMethod.JPEG;

   image.Settings.Compression = CompressionMethod.JPEG2000;
}
```

And to fix the colors you will need to change the colorspace to sRGB:

```C#
foreach (var image in images)
{
    image.ColorSpace = ColorSpace.sRGB;
}
```

**Reply by abbottmw** [reply]
Thanks YOU!   this helped a ton.   Got the pdf down to < 1MB and can go smaller if i figit with the quality of the images.
